java/src/jmri/jmrit/logixng/actions/ActionBlock.java

```diff
@@ -0,0 +1,607 @@
+        return _blockConstant;
+    }
+
+    public void setBlockMemory(@Nonnull String memoryName) {
+        assertListenersAreNotRegistered(log, "setBlockMemory");
+        MemoryManager memoryManager = InstanceManager.getDefault(MemoryManager.class);
+        Memory memory = memoryManager.getMemory(memoryName);
+        if (memory != null) {
+            setBlockMemory(memory);
+        } else {
+            removeBlockMemory();
+            log.warn("memory \"{}\" is not found", memoryName);
+        }
+    }
+
+    public void setBlockMemory(@Nonnull Memory memory) {
+        assertListenersAreNotRegistered(log, "setBlockMemory");
+        setBlockMemory(InstanceManager.getDefault(NamedBeanHandleManager.class)
+                .getNamedBeanHandle(memory.getDisplayName(), memory));
+    }
+
+    public void setBlockMemory(@Nonnull NamedBeanHandle<Memory> handle) {
+        assertListenersAreNotRegistered(log, "setBlockMemory");
+        _blockMemoryHandle = handle;
+        addRemoveVetoListener();
+    }
+
+    public NamedBeanHandle<Memory> getBlockMemory() {
+        return _blockMemoryHandle;
+    }
+
+    public void removeBlockMemory() {
+        assertListenersAreNotRegistered(log, "removeOtherMemory");
+        if (_blockMemoryHandle != null) {
+            _blockMemoryHandle = null;
+            addRemoveVetoListener();
+        }
+    }
+
+    private void addRemoveVetoListener() {
+        if ((_blockMemoryHandle != null)) {
+            InstanceManager.getDefault(MemoryManager.class).addVetoableChangeListener(this);
+        } else {
+            InstanceManager.getDefault(MemoryManager.class).removeVetoableChangeListener(this);
+        }
+    }
+
+    @Override
+    public void vetoableChange(java.beans.PropertyChangeEvent evt) throws java.beans.PropertyVetoException {
+        if ("CanDelete".equals(evt.getPropertyName())) { // No I18N
+            if (evt.getOldValue() instanceof Block) {
+                if (evt.getOldValue().equals(getBlock().getBean())) {
+                    PropertyChangeEvent e = new PropertyChangeEvent(this, "DoNotDelete", null, null);
+                    throw new PropertyVetoException(Bundle.getMessage("ActionBlock_BlockInUseVeto", getDisplayName()), e); // NOI18N
+                }
+            }
+            if (evt.getOldValue() instanceof Memory) {
+                if (evt.getOldValue().equals(getBlockMemory().getBean())) {
+                    PropertyChangeEvent e = new PropertyChangeEvent(this, "DoNotDelete", null, null);
+                    throw new PropertyVetoException(Bundle.getMessage("ActionBlock_MemoryInUseVeto", getDisplayName()), e); // NOI18N
+                }
+            }
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public Category getCategory() {
+        return Category.ITEM;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public boolean isExternal() {
+        return true;
+    }
+
+    private String getNewLock() throws JmriException {
```

Comment:
Should this be `getNewBlock()` instead of `getNewLock()`?

Reply:
It was in the action that I copied.  It appears to be related to Reference, LocalVariable and Formula.

Reply:
My fault. It comes from `ActionTurnoutLock` there the enum is `TurnoutLock`. I have copied that class and missed to rename the method. In `ActionTurnout`, that method is called `getNewState()`.

Looking at the code, I think this method should be `getNewDirectOperation()`.